zerver/lib/streams.py

```diff
@@ -691,7 +702,7 @@ def access_stream_by_name(
         user_profile,
         stream,
         error,
-        allow_realm_admin=allow_realm_admin,
+        allow_users_with_metadata_access=allow_users_with_metadata_access,
```

Comment:
I expect we'll find ourselves wanting a different final name for this; whether it's `require_content_access` here or maybe splitting to different names for the functions, like `access_stream_for_metadata` vs `access_stream_for_content`.

Reply:
Would it be ok to leave this as a followup?